### PR TITLE
Feature carrier immobalization

### PR DESCRIPTION
**Pull request by marcelkuipers**
This is a basic implementation of how carrier immobalization should work.
I made some changes to MoveBehaviour classes and collision as well, which is relevant for people working on other branches. That's why I am making another PR.

### CODE REVIEW

File: ContextDesktop/src/main/java/com/github/migi_1/Context/model/MainEnvironment.java [hunk 1]
```diff
@@ -176,15 +190,16 @@ private void initSpatials() {
...
 
         for (DamageDealer damageDealer : damageDealerGenerator.getObstacles()) {
-            damageDealer.collideWith(platform.getModel().getWorldBound(), results);
-            addDisplayable(damageDealer);
+            for (Entry<Entity, CollisionResults> entry : results.entrySet()) {
+                damageDealer.collideWith(entry.getKey().getModel().getWorldBound(), entry.getValue());
+                addDisplayable(damageDealer);
+            }
         }
         for (Path path : levelGenerator.getPathPieces(COMMANDER_LOCATION)) {
```
Comment (dVoorhout) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/model/MainEnvironment.java:201]: Not sure why this in the initialization, shouldn't collision be checked in the main loop?

Comment (marcelkuipers) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/model/MainEnvironment.java:201]: You are correct, this piece of code is redundant, so I deleted it.

File: ContextDesktop/src/main/java/com/github/migi_1/Context/model/entity/CarrierMoveBehaviour.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+     */
+    @Override
+    public void updateMoveVector() {
+        //corner location under the platform.
+        Vector3f destination = commander.getModel().getLocalTranslation().add(relativeLocation);
+        //become one frame closer to not being immobilised.
+        if (immobalised > 0) {
+            immobalised -= 1;
+        }
+        //when the carrier overshoots, put him in the right location.
```
Comment (dVoorhout) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/model/entity/CarrierMoveBehaviour.java:89]: immobilized\* ;p

Comment (marcelkuipers) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/model/entity/CarrierMoveBehaviour.java:89]: I guess I will need to redo my CPE xD

**Comment by NilsHullegien** [comment]
Looks good!

**Pull request merged**